src/iap.ts

```diff
@@ -513,23 +555,24 @@ const App = () => {
 
  */
 
-export const requestPurchase = ({
-  sku,
-  andDangerouslyFinishTransactionAutomaticallyIOS = false,
-  obfuscatedAccountIdAndroid,
-  obfuscatedProfileIdAndroid,
-  appAccountToken,
-  skus, // Android Billing V5
-  isOfferPersonalized = undefined, // Android Billing V5
-  quantity,
-  withOffer,
-}: RequestPurchase): Promise<ProductPurchase | void> =>
+export const requestPurchase = (
+  request: RequestPurchase,
+): Promise<ProductPurchase | void> =>
   (
     Platform.select({
       ios: async () => {
-        if (!sku) {
-          return Promise.reject(new Error('sku is required for iOS purchase'));
+        if (!('sku' in request)) {
```

Comment:
We use the type narrowing technique ([docs](https://www.typescriptlang.org/docs/handbook/2/narrowing.html#the-in-operator-narrowing)) to tell if this is a `RequestPurchaseIOS` or `RequestPurchaseAndroid`. Typescript knows that only `RequestPurchaseIOS` has the property `sku`, so knows that it's the iOS type after this check.